projects/storefrontlib/src/cms-components/user/login-form/login-form.component.ts

```diff
@@ -14,48 +15,66 @@ import { CustomFormValidators } from '../../../shared/utils/validators/custom-fo
   templateUrl: './login-form.component.html',
 })
 export class LoginFormComponent implements OnInit, OnDestroy {
-  sub: Subscription;
+  private tokenExists = false;
+  subs: Subscription;
   form: FormGroup;
 
   constructor(
     private auth: AuthService,
     private globalMessageService: GlobalMessageService,
     private fb: FormBuilder,
-    private authRedirectService: AuthRedirectService
+    private authRedirectService: AuthRedirectService,
+    @Optional() private winRef: WindowRef
```

Comment:
My mistake here. It should be `private winRef?: WindowsRef`.

Reply:
Please provide deprecation comment to the constructor to make `winRef` required in later releases.